javascript/ql/lib/semmle/javascript/frameworks/SQL.qll

```diff
@@ -267,6 +278,66 @@ private module Sqlite {
   }
 }
 
+/**
+ * Provides classes modeling the `sqlite` package.
+ */
+private module Sqlite {
+  /** Gets an expression that constructs or returns a Sqlite database instance. */
+  API::Node database() {
+    result = API::moduleImport("sqlite").getMember("open").getReturn().getPromised()
+  }
+
+  /** A call to a Sqlite query method. */
+  private class QueryCall extends DatabaseAccess, API::CallNode {
+    QueryCall() {
+      this = database().getMember(["all", "each", "exec", "get", "prepare", "run"]).getACall()
+    }
+
+    override DataFlow::Node getAResult() { result = this }
+
+    override DataFlow::Node getAQueryArgument() { result = this.getArgument(0) }
+  }
+
+  /** An expression that is passed to the `query` method and hence interpreted as SQL. */
+  class QueryString extends SQL::SqlString {
+    QueryString() { this = any(QueryCall qc).getAQueryArgument() }
+  }
+}
+
+/**
+ * Provides classes modeling the `better-sqlite3` package.
+ */
+private module BetterSqlite3 {
+  /**
+   * Gets an expression that constructs or returns a `better-sqlite3` database instance.
+   */
+  API::Node database() {
+    // initialDatabaseInstance is an instance of Database that constructed and instantiated in the first step of Database initialization,
+    // not from a return value of the other library functions
+    exists(API::Node initialDatabaseInstance |
+      initialDatabaseInstance =
+        [
+          API::moduleImport("better-sqlite3").getInstance(),
+          API::moduleImport("better-sqlite3").getReturn()
+        ]
+    |
+      result = [initialDatabaseInstance, initialDatabaseInstance.getMember("exec").getReturn()]
+    )
+  }
```

Comment:
```suggestion
  API::Node database() {
    result =
      [
        API::moduleImport("better-sqlite3").getInstance(),
        API::moduleImport("better-sqlite3").getReturn()
      ]
    or
    result = database().getMember("exec").getReturn()
  }
```

Calling `exec()` on a database gives you the same database.   
But you can keep calling `exec()` on the result of that, so your model needs to be recursive.

Reply:
And if you see the types of that package: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/better-sqlite3/index.d.ts 
There are a few other methods on databases that returns the database back again.

Reply:
It's OK to leave them for now.  

But I'm pretty sure we could just add `loadExtension` and `function` (similarly to `exec`), I doubt that they change the database in ways that are relevant for our model.